envisage/tests/test_egg_basket_plugin_manager.py

```diff
@@ -9,76 +9,57 @@
 # Thanks for using Enthought open source!
 """ Tests for the 'Egg Basket' plugin manager. """
 
+import contextlib
 import glob
+import os
 import shutil
-import sys
 import tempfile
 import unittest
 from os.path import basename, join
 
 import pkg_resources
 
 from envisage.egg_basket_plugin_manager import EggBasketPluginManager
-from envisage.tests.test_egg_based import build_egg
+from envisage.tests.support import (
+    BAD_PLUGIN_PACKAGES,
+    build_egg,
+    PLUGIN_PACKAGES,
+    restore_pkg_resources_working_set,
+    restore_sys_modules,
+    restore_sys_path,
+    temporary_directory,
+)
 
 
 class EggBasketPluginManagerTestCase(unittest.TestCase):
     """Tests for the 'Egg Basket' plugin manager."""
 
-    #### 'unittest.TestCase' protocol #########################################
-
     @classmethod
     def setUpClass(cls):
-        """
-        Create eggs for testing purposes.
-        """
-        cls.eggs_dir = tempfile.mkdtemp()
-        cls.bad_eggs_dir = tempfile.mkdtemp()
-
-        eggs_root_dir = pkg_resources.resource_filename(
-            "envisage.tests", "eggs"
+        cls.egg_cleanup_stack = contextlib.ExitStack()
+        cls.eggs_dir = os.fspath(
+            cls.egg_cleanup_stack.enter_context(temporary_directory())
         )
-        for egg_name in ["acme-bar", "acme-baz", "acme-foo"]:
-            build_egg(
-                egg_dir=join(eggs_root_dir, egg_name),
-                dist_dir=cls.eggs_dir,
-            )
-
-        bad_eggs_root_dir = pkg_resources.resource_filename(
-            "envisage.tests", "bad_eggs"
+        cls.bad_eggs_dir = os.fspath(
+            cls.egg_cleanup_stack.enter_context(temporary_directory())
         )
-        for egg_name in ["acme-bad"]:
-            build_egg(
-                egg_dir=join(bad_eggs_root_dir, egg_name),
-                dist_dir=cls.bad_eggs_dir,
-            )
+
+        # Build eggs
+        for package in PLUGIN_PACKAGES:
+            build_egg(package_dir=package, dist_dir=cls.eggs_dir)
+        for package in BAD_PLUGIN_PACKAGES:
+            build_egg(package_dir=package, dist_dir=cls.bad_eggs_dir)
 
     @classmethod
     def tearDownClass(cls):
```

Comment:
I'd prefer to use `addClassCleanup`, but that's not available until Python 3.8 and we're still supporting Python 3.7.